// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option objc_class_prefix = "TDS";

package tiramisudataserver;

// The greeting service definition.
service TiramisuDataServer {
  // Sends a greeting
  rpc GetTiramisuFunction (TiramisuFunctionName) returns (TiramisuFunction) {}
  rpc SaveTiramisuFunction (TiramisuFunction) returns (TiramisuFunctionName) {}
  rpc GetDatasetSize (Empty) returns (DatasetSize) {}
  rpc GetListOfFunctions (Empty) returns (TiramisuListOfFunctions) {}
}

// The request message containing the user's name.
message TiramisuFunctionName {
  string name = 1;
}

// The response message containing the greetings
message TiramisuFunction {
  string name = 1;
  string content = 2;
  string cpp = 3;
  bytes wrapper = 4;
}

message DatasetSize {
  int64 size = 1;
}

message Empty {
}

message TiramisuListOfFunctions {
  repeated string names = 1;
}